gtk4/src/subclass/widget.rs

```diff
@@ -181,18 +181,25 @@ impl<T: WidgetImpl> WidgetImplExt for T {
         unsafe {
             let data = T::type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkWidgetClass;
-            let f = (*parent_class)
-                .compute_expand
-                .expect("No parent class impl for \"compute_expand\"");
-            let mut h: i32 = hexpand_p.to_glib();
-            let mut v: i32 = vexpand_p.to_glib();
-            f(
-                widget.unsafe_cast_ref::<Widget>().to_glib_none().0,
-                &mut h,
-                &mut v,
-            );
-            *hexpand_p = from_glib(h);
-            *vexpand_p = from_glib(v);
+            let widget = widget.unsafe_cast_ref::<Widget>();
+            if let Some(f) = (*parent_class).compute_expand {
+                let mut h: i32 = hexpand_p.to_glib();
+                let mut v: i32 = vexpand_p.to_glib();
+                f(widget.to_glib_none().0, &mut h, &mut v);
+                if !widget.get_hexpand_set() {
```

Comment:
Ah I missed that these are readable from the outside. That makes it less annoying to handle this correctly as you can know when it's uninitialized and in that case you can just initialize it to `false` or so :)